backend/runner/taskrun/executor.go

```diff
@@ -474,6 +444,54 @@ func postMigration(ctx context.Context, stores *store.Store, activityManager *ac
 	}, nil
 }
 
+func isWriteBack(ctx context.Context, stores *store.Store, license enterpriseAPI.LicenseService, project *store.ProjectMessage, repo *api.Repository, task *store.TaskMessage, vcsPushEvent *vcsPlugin.PushEvent) (bool, error) {
+	if !license.IsFeatureEnabled(api.FeatureVCSSchemaWriteBack) {
+		return false, nil
+	}
+	if repo == nil || repo.SchemaPathTemplate == "" {
+		return false, nil
+	}
+	if task.Type == api.TaskDatabaseSchemaBaseline {
+		if strings.Contains(repo.BranchFilter, "*") {
+			return false, nil
+		}
+		return true, nil
+	}
+
+	if vcsPushEvent != nil {
```

Comment:
Why don't we write back if the `vcsPushEvent` is nil?

Reply:
We are using "pushEvent.Ref" in the writeBackLatestSchema().